### PR TITLE
Implement topic

### PR DESCRIPTION
**Pull request by meetmangukiya**

### CODE REVIEW

**Comment by meetmangukiya** [comment]
See room schema : https://developer.gitter.im/docs/rooms-resource

**Comment by meetmangukiya** [comment]
@zoni updated, yes that totally made sense, topic after initialization would've been constant that way, good catch :+1:

File: gitter.py [hunk 1]
```diff
@@ -190,7 +191,11 @@ def leave(self, reason=None):
 
     @property
     def topic(self):
-        return "TODO"  # TODO
+        json = self._backend.readAPIRequest('rooms', {"q": self.uri}).json()
+        for element in json:
+            if element.uri == self.uri:
+                return element.topic
+        return "Couldn't find :("
```
Comment (zoni) [inline gitter.py]: This code should raise an exception if the URI isn't present in the response, as it's supposed to be there.

File: gitter.py [hunk 2]
```diff
@@ -149,11 +149,12 @@ class GitterRoom(Room):
     def invite(self, *args) -> None:
         pass
 
-    def __init__(self, backend, idd, uri, name):
+    def __init__(self, backend, idd, uri, name, topic=None):
         self._backend = backend
         self._name = name
         self._uri = uri
         self._idd = idd
+        self._topic = topic
```
Comment (zoni) [inline gitter.py]: We're not making use of the `topic` parameter nor `self._topic` anymore, no need to store it.

**Comment by meetmangukiya** [comment]
@zoni updated

File: gitter.py [hunk 3]
```diff
@@ -190,7 +193,12 @@ def leave(self, reason=None):
 
     @property
     def topic(self):
-        return "TODO"  # TODO
+        json = self._backend.readAPIRequest('rooms', {"q": self.uri}).json()
+        for element in json:
+            if element.uri == self.uri:
+                return element.topic
+            else:
+                raise RoomNotFoundError
```
Comment (zoni) [inline gitter.py]: I don't believe it's good practice to raise exceptions without a message. I would suggest the following: 

```python
raise RoomNotFoundError("Cannot find the room '{}'".format(self.name))
```

File: gitter.py [hunk 4]
```diff
@@ -190,7 +193,12 @@ def leave(self, reason=None):
 
     @property
     def topic(self):
-        return "TODO"  # TODO
+        json = self._backend.readAPIRequest('rooms', {"q": self.uri}).json()
+        for element in json:
+            if element.uri == self.uri:
```
Comment (zoni) [inline gitter.py]: What happens if `element` doesn't have an attribute `uri`?

File: gitter.py [hunk 5]
```diff
@@ -307,7 +315,8 @@ def rooms(self):
         for json_room in json_rooms:
             if not json_room['oneToOne']:
                 log.debug("found room %s (%s)" % (json_room['name'], json_room['uri']))
-                rooms.append(GitterRoom(self, json_room['id'], json_room['uri'], json_room['name']))
+                rooms.append(GitterRoom(self, json_room['id'], json_room['uri'],
+                                        json_room['name'], json_room['topic']))
```
Comment (zoni) [inline gitter.py]: `GitterRoom` doesn't have a 4th parameter (I believe you forgot to remove the `json_room['topic']` addition which was added in a previous iteration).

File: gitter.py (hunk 4 above)
Comment (meetmangukiya) [inline gitter.py]: element will always the attribute URI, it is the schema: See Room Schema section https://developer.gitter.im/docs/rooms-resource

File: gitter.py (hunk 5 above)
Comment (meetmangukiya) [inline gitter.py]: true :+1:

**Comment by zoni** [comment]
Thanks. :+1:

**Pull request merged**